services/QuillLMS/spec/controllers/teachers/progress_reports_controller_spec.rb

```diff
@@ -243,4 +243,17 @@
       })
     end
   end
+
+  describe '#building_ai_knowledge_unit_templates' do
+    before do
+      allow(controller).to receive(:course_with_assignment_data).and_return([])
+    end
+
+    it 'renders json with a unit_templates key' do
+      get :building_ai_knowledge_unit_templates
```

Comment:
This is pretty minor, but you could move this `get` call into the `before` block and get the same behavior.  I think that's slightly easier to expand if we ever added more specs here.  Still, it doesn't really change anything, so take or leave it.